api/accounts/views.py

```diff
@@ -1,17 +1,93 @@
+from rest_framework import generics, permissions
+from rest_framework.response import Response
+from knox.models import AuthToken
+from .serializers import UserSerializer, RegisterSerializer, LoginSerializer,ProfileSerializer
+from .models import Profile, CustomUser
 
-from django.shortcuts import render, redirect
-from rest_framework import generics
-from .serializers import ProfileSerializer
-from django.contrib.auth.models import User
-from .models import Profile
 
+class RegisterAPI(generics.GenericAPIView):
+    serializer_class = RegisterSerializer
 
-class ProfileList(generics.ListCreateAPIView):
+    def post(self, request, *args, **kwargs):
+        profile_data = request.data.pop('profile')
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+        profile = Profile.objects.create(user=user, **profile_data)
+        
+        return Response({
+            "user": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": AuthToken.objects.create(user)[1]
+        })
+        
+# Example json input for RegisterAPI^^^
+
+# {
+#     "email": "bigguy@gmail.com",
+#     "password": "airtravel",
+#     "profile": {
+#         "image": "http://127.0.0.1:8000/media/default.jpg",
+#         "first_name": "Mike",
+#         "last_name": "J",
+#         "city_of_residence": "Chicago",
+#         "age": 50,
+#         "dream_destination": "Chicago",
+#         "bio": "I hang out with loony tunes",
+#         "instagram_url": "",
+#         "pinterest_url": "",
+#         "facebook_url": ""
+#     }
+# }
```

Comment:
This could go in documentation. We only want documenting-style comments in the codebase though